### PR TITLE
Fix #7183 - Array bounds error in SizePIU

### LINKED ISSUE #7183: Array bounds error in SizePIU
Issue overview
--------------
User reported several types of unexplained error from the same file - different results on consecutive runs. Running with v9.1 debug build reveals an array bounds error [here](https://github.com/NREL/EnergyPlus/blob/f88eb67c7fc1539d9827bbde166762af34c833ee/src/EnergyPlus/PoweredInductionUnits.cc#L1464) in SizePIU. This line should be 
```
TermUnitSizing(CurTermUnitSizingNum).DesHeatingLoad = DesCoilLoad;
```

Note that this file fails on an assert in [DataSurfaces](https://github.com/NREL/EnergyPlus/blob/f88eb67c7fc1539d9827bbde166762af34c833ee/src/EnergyPlus/DataSurfaces.cc#L606), so need to comment that out to test this in debug build. Not sure what the issue is with that.

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus v9.1.0-f88eb67c7f
 - Helpdesk ticket 13180

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added in EnergyPlusDevSupport\DefectFiles, use San Antonio TX TMY3 (but any weather would trigger the issue).
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Fix #7183 - Array bounds error in SizePIU. 

Fix was straightforward, but started by testing with user file, and added a Unit test that would reproduce a similar array out of bound error.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by jmarrec** [comment]
Ok I cleaned up my unit test, and I cleaned up my commits (and reordered them) hence the force push. Ready for review now.

**Review by rraustad — APPROVED** [review]
Change looks accurate. New unit test calls sizing function to verify change. No redundancy in unit test (i.e., no other unit test calls sizing).

File: src/EnergyPlus/PoweredInductionUnits.cc [hunk 1]
```diff
@@ -1461,7 +1461,7 @@ namespace PoweredInductionUnits {
             TermUnitSizing(CurTermUnitSizingNum).MinFlowFrac = PIU(PIUNum).MinPriAirFlowFrac;
             TermUnitSizing(CurTermUnitSizingNum).MaxHWVolFlow = PIU(PIUNum).MaxVolHotWaterFlow;
             TermUnitSizing(CurTermUnitSizingNum).MaxSTVolFlow = PIU(PIUNum).MaxVolHotSteamFlow;
-            TermUnitSizing(CurZoneEqNum).DesHeatingLoad = DesCoilLoad; // coil report
+            TermUnitSizing(CurTermUnitSizingNum).DesHeatingLoad = DesCoilLoad; // coil report
```
Comment (Myoldmopar) [inline src/EnergyPlus/PoweredInductionUnits.cc:1464]: Seems right...

File: tst/EnergyPlus/unit/PoweredInductionUnits.unit.cc [hunk 2]
```diff
@@ -523,3 +523,55 @@ TEST_F(EnergyPlusFixture, SeriesPIUTest1)
     DataZoneEnergyDemands::ZoneSysEnergyDemand.deallocate();
     DataZoneEnergyDemands::CurDeadBandOrSetback.deallocate();
 }
+
+
+// cf: https://github.com/NREL/EnergyPlus/issues/7183
+TEST_F(EnergyPlusFixture, PIUArrayOutOfBounds) {
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/PoweredInductionUnits.unit.cc:529]: Good, succinct unit test.  Don't really need the comment above stating what it is for, we can track that through our history, but no point in re-running CI to remove the comment.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Built and tested locally, defect file runs to completion.  Merging.

**Pull request merged**